Dockerfile.develop

```diff
@@ -90,16 +90,23 @@ RUN true\
 # won't invalidate earlier download layers
 RUN go mod download
 
-# Install controller-gen
+# Export the Go binary path for controller-gen and ginkgo CLIs
+ENV PATH $HOME/go/bin:$PATH
+
+# Install controller-gen to generate util code and Kubernetes YAMLs for API changes
+RUN true \
+    && go install sigs.k8s.io/controller-tools/cmd/controller-gen@${CONTROLLER_GEN_VERSION} \
+    && controller-gen --version \
+    && true
+
+# Install the Ginkgo test framework
 RUN true \
-    && mkdir /tmp/controller-gen-tmp && cd /tmp/controller-gen-tmp \
-    && go mod init tmp \
-    && go get sigs.k8s.io/controller-tools/cmd/controller-gen@${CONTROLLER_GEN_VERSION} \
-    && rm -rf /tmp/controller-gen-tmp \
+    && go install github.com/onsi/ginkgo/v2/ginkgo \
+    && ginkgo version \
     && true
 
-RUN go install github.com/onsi/ginkgo/v2/ginkgo
-ENV PATH=/root/go/bin/:$PATH
+# For GitHub Action 'lint', work around error "detected dubious ownership in repository at '/workspace'"
+RUN git config --system --add safe.directory /workspace
```

Comment:
https://github.com/kserve/modelmesh-serving/actions/runs/5126882567/jobs/9221907299#step:5:14

```
Run ./scripts/develop.sh make fmt
./scripts/fmt.sh || (echo "Linter failed: $?"; git status; exit 1)
An error has occurred: FatalError: git failed. Is it installed, and are you in a Git repository directory?
Check the log at /root/.cache/pre-commit/pre-commit.log
Pre-commit linter failed, but it may have automatically formatted your files.
Check your changed files and/or manually fix the errors above then stage and commit.
Linter failed: 1
fatal: detected dubious ownership in repository at '/workspace'
To add an exception for this directory, call:

	git config --global --add safe.directory /workspace
make: *** [Makefile:139: fmt] Error 1
Error: Process completed with exit code 2.
```